api/src/controllers/referent.js

```diff
@@ -928,7 +931,18 @@ router.get("/young/:id", passport.authenticate("referent", { session: false, fai
       const structure = await StructureModel.findById(application.structureId);
       applications.push({ ...application._doc, structure: structure ? serializeStructure(structure, req.user) : null });
     }
-    return res.status(200).send({ ok: true, data: { ...data._doc, applications } });
+
+    let etablissement = undefined;
+    let classe = undefined;
+
+    if (data.source === YOUNG_SOURCE.CLE) {
+      etablissement = await EtablissementModel.findById(data.etablissementId);
+      classe = await ClasseModel.findById(data.classeId);
+      //Can throw 404 because when preinsciption for young CLE is finished, we already this information
+      if (!etablissement || !classe) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
```

Comment:
pas sur de comprendre le commentaire mais why not

Reply:
En gros si un jeune a pas de classe ni établissement c'est pas logique ^^ -> donc 404